### DISCUSSION TITLE
Unable to add array to model but if I convert the data with toString(), I am able to add it

### DISCUSSION THREAD

**Discussion by jhoangqm** [prisma/prisma#17064] [Q&A] [answered]
Hello everyone,

Tech Stack:
Next.js
Postgresql
Prisma
Javscript

I am running into an issue where I am trying to seed data to the database but I am unable to proceed.

When I don't use the .toString(), I get errors saying that it's expecting the data to be a list of strings or null.

My real question is, is it possible to store an array? A list of string seems to work but it is not what I am looking for.

Any tips or pointers is appreciated!

Here's the lines of code:

```
model Models {
  id      Int     @id @default(autoincrement())
  model   String[]
  makes   Makes   @relation(fields: [makesId], references: [id])
  makesId Int

  @@map("models")
}

 **The problem lies here**

  for (const carModel of carModels) {
    console.log("carModel.model is: ", carModel.model);
    await prisma.models.create({
      data: {
        id: carModel.modelId,
        model: carModel.model.toString(),
        makesId: carModel.makesId,
      },
    });
  }

**The data I am trying to seed**
const carModels = [
  {
    modelId: 1,
    model: [
      "CL",
      "ILX",
      "Integra",
      "Legend",
      "MDX",
      "MDX Sport Hybrid",
      "NSX",
      "RDX",
      "RL",
      "RLX",
      "RLX Sport Hybrid",
      "RSX",
      "SLX",
      "TL",
      "TLX",
      "TSX",
      "Vigor",
      "ZDX",
    ],
    makesId: 1,
  },
  ]
```

Here's the full error message:
carModel.model is:  [
  'CL',               'ILX',
  'Integra',          'Legend',
  'MDX',              'MDX Sport Hybrid',
  'NSX',              'RDX',
  'RL',               'RLX',
  'RLX Sport Hybrid', 'RSX',
  'SLX',              'TL',
  'TLX',              'TSX',
  'Vigor',            'ZDX'
]
PrismaClientUnknownRequestError: 
Invalid `prisma.models.create()` invocation in
/home/asdasd/asdasd/asdasd/client/prisma/seed.ts:24:25

  21 
  22 for (const carModel of carModels) {
  23   console.log("carModel.model is: ", carModel.model);
→ 24   await prisma.models.create(
Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: ToSql(1), cause: Some(Error { kind: ConversionError("Couldn't serialize value `Some([Text(Some(\"CL,ILX,Integra,Legend,MDX,MDX Sport Hybrid,NSX,RDX,RL,RLX,RLX Sport Hybrid,RSX,SLX,TL,TLX,TSX,Vigor,ZDX\"))])` into a `text`. Value is a list but `text` is not."), original_code: None, original_message: None }) }) })
    at RequestHandler.handleRequestError (Removed due to secrecy)
    at RequestHandler.handleAndLogRequestError (Removed due to secrecy)
    at RequestHandler.request (Removed due to secrecy)
    at async PrismaClient._request (Removed due to secrecy) {
  clientVersion: '4.8.0',
  batchRequestIdx: undefined
}

Please let me know if you need more information

**Comment by jhoangqm** [comment, marked as answer]
Question can be closed!

It was a dumb mistake on my part, the data seeded wasn't done properly. I needed to change the data structure to suit my needs.

Original car data was changed to the following:

```
const carModels = [
  {
    modelId: 1,
    makesId: 1,
    model: "CL",
  },
  {
    modelId: 2,
    makesId: 1,
    model: "ILX",
  },
  {
    modelId: 3,
    makesId: 1,
    model: "Integra",
  },
  {
    modelId: 4,
    makesId: 1,
    model: "Legend",
  },
  {
    modelId: 5,
    makesId: 1,
    model: "MDX",
  },
  {
    modelId: 6,
    makesId: 1,
    model: "MDX Sport Hybrid",
  },
  {
    modelId: 7,
    makesId: 1,
    model: "NSX",
  },
  {
    modelId: 8,
    makesId: 1,
    model: "RDX",
  },
  {
    modelId: 9,
    makesId: 1,
    model: "RL",
  },
  {
    modelId: 10,
    makesId: 1,
    model: "RLX",
  },
  {
    modelId: 11,
    makesId: 1,
    model: "RLX Sport Hybrid",
  },
  {
    modelId: 12,
    makesId: 1,
    model: "RSX",
  },
  {
    modelId: 13,
    makesId: 1,
    model: "SLX",
  },
  {
    modelId: 14,
    makesId: 1,
    model: "TL",
  },
  {
    modelId: 15,
    makesId: 1,
    model: "TLX",
  },
  {
    modelId: 16,
    makesId: 1,
    model: "TSX",
  },
  {
    modelId: 17,
    makesId: 1,
    model: "Vigor",
  },
  {
    modelId: 18,
    makesId: 1,
    model: "ZDX",
  },
];```

**Reply by pantharshit00** [reply]
I am glad that you were able to figure this out. Feel free to reach out again!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
